### PR TITLE
fix: assets with special characters

### LINKED ISSUE #121: Does not handle Japanese character
FIrst of all, thank you for this brillant package!
However, I had some problems getting started because this package does not seem to handle some characters (correctly).

I have a file called `う.wav` which cannot be played. I think this is not a problem but there is on error message (or anything) indicating that the name is the problem, the sound just does not play. Renaming the file to `u.wav` resolved the issue

I am converting from `just_audio` to this package (in just audio the name works) and this took me quite some time to figure out, therefore, a note somewhere would be nice (in my opinion).

### PR DESCRIPTION
**Pull request by alexmercerind**
Closes #121.

### CODE REVIEW

**Comment by birros** [comment]
I recommend adding a small unit test to cover this case... and the following potential cases...
Even if it takes a bit more time to fix.

**Comment by alexmercerind** [comment]
I really agree for the tests. We seriously need unit tests inside package:media_kit.

There's a lot to cover at the moment. The rendering size is still well tested.

**Comment by birros** [comment]
I added a unit test, I don't know the Dart / Flutter conventions, I did what I thought was right.

**Review by birros — APPROVED** [review]

**Comment by alexmercerind** [comment]
Converting to draft for now based on the last comment on issue thread.

File: media_kit/lib/src/models/media.dart [hunk 1]
```diff
@@ -46,58 +47,59 @@ class Media extends Playable {
...
         );
+      } else if (Platform.isAndroid) {
+        asset = path.normalize(AndroidAssetLoader.instance.load(key));
+      } else {
+        throw UnimplementedError(
+          '$_kAssetScheme is not supported on ${Platform.operatingSystem}',
+        );
+      }
+      if (!File(asset).existsSync_()) {
+        throw Exception('Unable to load asset: $asset');
```
Comment (alexmercerind) [inline media_kit/lib/src/models/media.dart:100]: Now throwing an exception if an asset is not located.

**Pull request merged**